experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -509,10 +472,10 @@ static int ws_eink_spi_probe(struct spi_device *spi)
 	info->fix.ywrapstep	= 0;
 	info->fix.line_length	= width * bpp / 8;
 
-	info->var.xres			= width;
+	info->var.xres			= roundup(width * bpp, 8) / bpp;
```

Comment:
As mentioned elsewhere, fixing the 122/128 issue is out of scope for this commit.  However, when this *is* fixed, I think this is the wrong solution.  I believe that ultimately `xres` should be `122` and `xres_virtual` should be `128`.